tests/trx_finality_status_test.py

```diff
@@ -0,0 +1,227 @@
+        assert "head_number" in status and "head_id" in status and "head_timestamp" in status, \
+            f"ERROR: getTransactionStatus is missing \"head_number\", \"head_id\", and \"head_timestamp\".\nstatus: {json.dumps(status, indent=1)}"
+
+        assert "irreversible_number" in status and "irreversible_id" in status and "irreversible_timestamp" in status, \
+            f"ERROR: getTransactionStatus is missing \"irreversible_number\", \"irreversible_id\", and \"irreversible_timestamp\".\nstatus: {json.dumps(status, indent=1)}"
+
+        assert not knownTrx or "expiration" in status, \
+            f"ERROR: getTransactionStatus is missing \"expiration\".\nstatus: {json.dumps(status, indent=1)}"
+
+        assert "earliest_tracked_block_id" in status and "earliest_tracked_block_number" in status, \
+            f"ERROR: getTransactionStatus is missing \"earliest_tracked_block_id\" and \"earliest_tracked_block_number\".\nstatus: {json.dumps(status, indent=1)}"
+
+    validate(status[0])
+
+    # since the transaction is traveling opposite to the flow of blocks (prodNodes[0] then prodNodes[1] etc) once the flow of blocks has progressed to testNode's
+    # nearest neighbor, it will at least be in its block if not one of the earlier ones
+    testNode.waitForProducer("defproducerb")
+    testNode.waitForProducer("defproducerc")
+
+    status.append(testNode.getTransactionStatus(transId))
+    state = getState(status[1])
+    assert state == inBlockState, f"ERROR: getTransactionStatus never returned a \"{inBlockState}\" state"
+
+    validateTrxState(status[1], present=True)
+
+    validate(status[1])
+
+    assert status[0]["head_number"] < status[1]["head_number"], \
+        f"ERROR: Successive calls to getTransactionStatus should return increasing values for \"head_number\".\n1st status: {json.dumps(status[0], indent=1)}\n\n2nd status: {json.dumps(status[1], indent=1)}"
+
+    block_number = status[1]["block_number"]
+    assert testNode.waitForIrreversibleBlock(block_number, timeout=180), \
+        f"ERROR: Failed to advance irreversible block to {block_number}. \nAPI Node info: {json.dumps(prod0.getInfo(), indent=1)}\n\nProducer info: {json.dumps(testNode.getInfo(), indent=1)}"
+
+    retStatus=testNode.getTransactionStatus(transId)
+    Print(f"retStatus: {retStatus}")
+    state = getState(retStatus)
+    assert state == irreversibleState, \
+        f"ERROR: Successive calls to getTransactionStatus should have resulted in eventual \"{irreversibleState}\" state." + \
+        f"\n1st status: {json.dumps(status[0], indent=1)}\n\n2nd status: {json.dumps(status[1], indent=1)}" + \
+        f"\n\nfinal status: {json.dumps(retStatus, indent=1)}"
+    validate(retStatus)
+
+    leeway=4
+    assert testNode.waitForBlock(blockNum=startingBlockNum+(successDuration*2),timeout=successDuration+leeway)
+
+    recentBlockNum=testNode.getBlockNum()
+    retStatus=testNode.getTransactionStatus(transId)
+    Print(f"retStatus: {retStatus}")
```

Comment:
Debug output left over?  These don't seem particularly useful for the normal progress of the test.